### PR TITLE
PLT-5880 Channel drawer UI

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Update the Channel Drawer UI including the animation to open the drawer adding opacity to the main view.

Partially moved the account settings from the right menu to the channel drawer just to have the functionality there, the right menu drawer will completely disappear once https://mattermost.atlassian.net/browse/PLT-5930

Removed the line separator for members outside of the team and added functionality for the GM channel.

Also fixes a bad merge for opening Modal views

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-5880
https://mattermost.atlassian.net/browse/PLT-5781

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] All new/modified APIs include changes to [mattermost-redux](https://github.com/mattermost/mattermost-redux) (https://github.com/mattermost/mattermost-redux/pull/10)
- [x] Has UI changes

#### Device Information
This PR was tested on: IOS 10.2.1 iPhone 6s, Android 6.0.1 Samsung J5

#### Screenshots
IOS
img_0336

Android
screenshot_20170319-210753

### CODE REVIEW

File: app/components/channel_drawer_list/channel_drawer_list.js [hunk 1]
```diff
@@ -13,13 +13,16 @@ import {
     View
 } from 'react-native';
 import {injectIntl, intlShape} from 'react-intl';
+import AwesomeIcon from 'react-native-vector-icons/FontAwesome';
+import MaterialIcon from 'react-native-vector-icons/MaterialIcons';
```
Comment (hmhealey) [inline app/components/channel_drawer_list/channel_drawer_list.js]: Why are we mixing Font Awesome and Material icons? I think we should stick with Font Awesome to keep everything consistent

File: app/components/channel_drawer_list/channel_drawer_list.js [hunk 2]
```diff
@@ -326,90 +312,91 @@ class ChannelDrawerList extends Component {
...
-        }
-
         this.firstUnreadChannel = null;
         this.lastUnreadChannel = null;
         this.findUnreadChannels(data);
 
         return data;
     };
 
+    renderSectionAction = (styles, action) => {
```
Comment (hmhealey) [inline app/components/channel_drawer_list/channel_drawer_list.js]: Would these (renderSectionAction, renderDivider, renderTitle) maybe make more sense as separate components?

File: app/components/channel_drawer_list/channel_drawer_list.js (hunk 1 above)
Comment (enahum) [inline app/components/channel_drawer_list/channel_drawer_list.js]: Ask the designers hehehehehe

File: app/components/channel_drawer_list/channel_drawer_list.js (hunk 2 above)
Comment (enahum) [inline app/components/channel_drawer_list/channel_drawer_list.js]: I don't see where can we re-use them, but I could 0/5 on this, you?

Comment (hmhealey) [inline app/components/channel_drawer_list/channel_drawer_list.js]: I'm thinking more for encapsulation and to make it a bit more obvious what each of the 5 parameters for renderTitle are. But as you said, they won't be reused so it doesn't really matter

**Pull request merged**